docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -0,0 +1,1938 @@
+name, and are built at 1.16 or above. For packages built at LF 1.15 or
+lower, the templates will not be identifiable via a package name, and a
+package id must be used.
+
+Note: template ids in query results will always use a package id. This
+allows us to distinguish the source of a particular contract. This means
+that if you use a template with a package name in the request, you can
+no longer expect the template ids in the result to exactly match the
+input template id.
+
+Package ID selection preference: preferences apply to JSON API where you
+can specify your preferred selection of package versions.
+
+PQS
+---
+
+TODO: Update this depending on what happens with PQS.
+
+To match the package-name changes to the Ledger API, PQS has added
+support for package-name queries of contracts. All queries that take a
+Daml identity in the form ``<package-id>:<module-name>:<template-name>`` now
+take a package-name in place of package-id. Note that this differs from
+the Ledger API in that the `#` prefix is not required for PQS. Queries for
+package-names will return all versions of a given contract, alongside
+the package-version and package-id for each contract.
+
+Given that PQS uses a document-oriented model for ledger content
+(JSONB), extensions to contract payloads are handled simply by returning
+the additional data in the blob.
+
+-  Should you ever need to reconstruct the old experience querying one
+   specific version, simply use a filter predicate in the SQL. eg.
+   ``SELECT \* FROM active('my_package:My.App:MyTemplate') WHERE package_id = 'deadbeef…'``.
+
+Daml Shell
+----------
+
+Daml Shell builds on PQS by providing a shell interface to inspect the
+ledger using package-name to view all versions of contracts, in an
+integrated way.
+
+Daml-Script-Beta
+----------------
+
+Daml 2.9 introduces a new version of Daml Script, which can be used by
+depending on ``daml-script-beta`` in your ``daml.yaml``, as you will have seen
+in `Writing your first upgrade <#writing-your-first-upgrade>`__. Only this version of Daml Script
+supports upgrades over the ledger-api.
+
+All commands and queries in this version of daml script will now use
+upgrades/downgrades automatically, to ensure that the correct versions
+of choices are exercises, and correct payloads are returned.
+
+The following additional functionality is available for more advanced
+uses of SCU.
+
+**Exact commands**
+
+Each of the 4 submission commands now have an “exact” variant, of the
+forms ``createExactCmd``, ``exerciseExactCmd``, ``exerciseByKeyExactCmd`` and
+``createAndExerciseExactCmd``. These commands will force the participant to
+use the exact version of the package that your script uses, this is most
+useful when you want to be absolutely certain of the choice code you are
+calling. Note that exact and non-exact commands can be mixed in the same
+submission.
+
+**Package Preference**
+
+TODO: Decide whether or not to include this
```

Comment:
Need a verdict here, probably going to exclude it, could always add for 2.9.1

@basvangijzel-DA

Reply:
I would exclude. Based on my recent conversations with @meiersi-da, using Daml Script for upgrade testing, particularly on the integration test level, is not considered high priority. Given that package preference is really more for integration testing I would just leave it out until we have support in the IDE ledger.